proxy/src/bin/pg_sni_router.rs

```diff
@@ -141,25 +141,35 @@ async fn task_main(
         tokio::select! {
             accept_result = listener.accept() => {
                 let (socket, peer_addr) = accept_result?;
-                info!("accepted postgres client connection from {peer_addr}");
-
                 let session_id = uuid::Uuid::new_v4();
+
+                info!(
+                    session_id = ?session_id,
+                    "accepted postgres client connection from {peer_addr}",
+                );
+
                 let tls_config = Arc::clone(&tls_config);
                 let dest_suffix = Arc::clone(&dest_suffix);
 
                 connections.spawn(
                     async move {
-                        info!("spawned a task for {peer_addr}");
+                        info!(
+                            session_id = ?session_id,
+                            "spawned a task for {peer_addr}",
+                        );
 
                         socket
                             .set_nodelay(true)
                             .context("failed to set socket option")?;
 
                         handle_client(dest_suffix, tls_config, session_id, socket).await
                     }
-                    .unwrap_or_else(|e| {
+                    .unwrap_or_else(move |e| {
```

Comment:
Instead I think we should add a to capture everything this call to the session.

The `unwrap_or_else` is on a strange position, but not changed here, I don't think I understand it's function, need to look at this in an editor :)

Reply:
Aah, silly me, it's a `FutureExt::unwrap_or_else`.